packages/devtools_app/lib/src/code_size/code_size_attribution.dart

```diff
@@ -0,0 +1,310 @@
+        );
+      },
+    );
+  }
+
+  Widget _buildFromTable() {
+    return FlatTable<CallGraphNode>(
+      key: CallGraphView.fromTableKey,
+      columns: [_fromColumn],
+      data: node.pred,
+      keyFactory: (CallGraphNode node) => ValueKey<CallGraphNode>(node),
+      onItemSelected: _selectMainNode,
+      sortColumn: _fromColumn,
+      sortDirection: SortDirection.descending,
+    );
+  }
+
+  Widget _buildToTable() {
+    return FlatTable<CallGraphNode>(
+      key: CallGraphView.toTableKey,
+      columns: [_toColumn],
+      data: node.succ,
+      keyFactory: (CallGraphNode node) => ValueKey<CallGraphNode>(node),
+      onItemSelected: _selectMainNode,
+      sortColumn: _toColumn,
+      sortDirection: SortDirection.descending,
+    );
+  }
+
+  Widget _buildMainNode() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const Padding(
+          padding: EdgeInsets.symmetric(horizontal: densePadding),
+          child: Icon(Icons.arrow_forward),
+        ),
+        Tooltip(
+          waitDuration: tooltipWait,
+          message: node.data.toString(),
+          child: Container(
+            padding: const EdgeInsets.all(densePadding),
+            child: Text(
+              node.display,
+              overflow: TextOverflow.ellipsis,
+            ),
+          ),
+        ),
+        const Padding(
+          padding: EdgeInsets.symmetric(horizontal: densePadding),
+          child: Icon(Icons.arrow_forward),
+        ),
+      ],
+    );
+  }
+
+  void _selectMainNode(CallGraphNode node) {
+    setState(() {
+      this.node = node;
+    });
+  }
+}
+
+class FromColumn extends ColumnData<CallGraphNode> {
+  FromColumn() : super('From');
+
+  @override
+  String getValue(CallGraphNode dataObject) => dataObject.display;
+}
+
+class ToColumn extends ColumnData<CallGraphNode> {
+  ToColumn() : super('To');
+
+  @override
+  ColumnAlignment get alignment => ColumnAlignment.right;
+
+  @override
+  String getValue(CallGraphNode dataObject) => dataObject.display;
+}
+
+class DominatorTree extends StatelessWidget {
+  DominatorTree({
+    @required this.dominatorTreeRoot,
+    @required this.selectedNode,
+  });
+
+  static const dominatorTreeTableKey = Key('DominatorTree - table');
+
+  final DominatorTreeNode dominatorTreeRoot;
+
+  final CallGraphNode selectedNode;
+
+  final _packageColumn = _PackageColumn();
+
+  @override
+  Widget build(BuildContext context) {
+    _expandToSelected();
+    // TODO(kenz): programmatically select [selectedNode] in the table.
+    return TreeTable<DominatorTreeNode>(
+      key: dominatorTreeTableKey,
+      dataRoots: [dominatorTreeRoot],
+      columns: [_packageColumn],
+      treeColumn: _packageColumn,
+      keyFactory: (node) => PageStorageKey<String>('${node.callGraphNode.id}'),
+      sortColumn: _packageColumn,
+      sortDirection: SortDirection.descending,
+      autoExpandRoots: true,
+    );
+  }
+
+  void _expandToSelected() {
```

Comment:
it would be better to do this on selection change rather than the build method.
modifying state in the build method can cause problems as build methods shouldn't ever have side effects for hot reload and flutter in general to work consistently.